hack/test/unit: run in the right module when TESTDIRS is used

akerouanton opened the pull request:
**- What I did**

Since 'api/' and 'client/' are separate Go modules, tests need to be run separately in each module. Commit https://github.com/moby/moby/commit/900a0516de7e9261474c67ebd39f8ddd652060fa changed the hack/test/unit script to account for that.

But since that commit, if that script is invoked with TESTDIRS set, it will try every module instead of locating the one containing TESTDIRS.

thaJeztah commented:
@akerouanton I tried to give it a go with my suggestion; pushed as a separate PR because there's quite some diffs with your PR;

- https://github.com/moby/moby/pull/50721

(also happy to rebase it on top of your commit to see the diff with your changes)

akerouanton committed:
hack/test/unit: run in the right module when TESTDIRS is used

Since 'api/' and 'client/' are separate Go modules, tests need to be run
separately in each module. Commit 900a051 changed the hack/test/unit
script to account for that.

But since that commit, if that script is invoked with TESTDIRS set, it
will try every module instead of locating the one containing TESTDIRS.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>

thaJeztah commented:
superseded by https://github.com/moby/moby/pull/50721
